design/ceph-nfs-ganesha.md

```diff
@@ -0,0 +1,166 @@
+# Ceph NFS-Ganesha CRD
+
+[NFS-Ganesha] is a user space NFS server that is well integrated with [CephFS]
+and [RGW] backends. It can export Ceph's filesystem namespaces and Object
+gateway namespaces over NFSv4 protocol.Rook already orchestrates Ceph filesytem
+and Object store (or RGW) on Kubernetes. It can be extended to orchestrate
+NFS-Ganesha server daemons as highly available and scalable NFS gateway pods
+to the Ceph filesytem and Object Store. This will allow NFS client applications
+to use the Ceph filesytem and object store setup by rook.
+
+## Design of Ceph NFS-Ganesha CRD
+
+The NFS-Ganesha server settings will be exposed to rook as a Custom Resource
+Definition (CRD). Creating the nfs-ganesha CRD will launch NFS Ganesha server
+pods that will be custom configured for CephFS and RGW backends (e.g., Ganesha
+caching turned off). The servers' exports and client recovery data will be
+stored in Ceph RADOS pools to allow the Ganesha server pods to scale and be
+highly available. And the servers will have stable IP addresses.
+
+### Prerequisities
+
+- A running rook Ceph filesystem or object store, whose namespaces will be
+  exported by the Ganesha servers.
+  e.g.,
+  ```
+  kubectl create -f filesystem.yaml
+  ```
+
+- An existing RADOS pool (e.g., CephFS's data pool) or a pool created from
+  the [Rook toolbox] that will  store the Ganesha exports and NFS client
+  recovery data.
+
+- A RADOS object in the pool that will store the Ganesha exports, and an
+  initial export stored in the object that the server will start off exporting.
+  ```
+  # cat << EOF > ganesha-export.conf
+  EXPORT
+  {
+        Export_ID=100;
+        Protocols = 4;
+        Transports = TCP;
+        Path = /;
+        Pseudo = /cephfs/;
+        Access_Type = RW;
+        Attr_Expiration_Time = 0;
+        Delegations = R;
+        Squash = No_root_squash;
+        FSAL {
+                Name = CEPH;
+        }
+  }
+  EOF
+  # rados -p myfs-data0 put ganesha-export ganesha-export.conf
+  ```
```

Comment:
Is there some reason to keep this config block in RADOS, but have the rest stored in a ganesha.conf file? Just wondering why it was decided to split the config that way. Maybe we should store more of the config in RADOS, and just keep the part that really has to be per-server in the local config file?